src/robotFactory.js

```diff
@@ -1,10 +1,87 @@
 'use strict';
 
-class BaseRobot {}
+class BaseRobot {
+  constructor(name, weight, coords = {
+    x: 0,
+    y: 0,
+  }, chipVersion) {
+    this.name = name;
+    this.weight = weight;
 
-class FlyingRobot {}
+    this.coords = {
+      x: 0,
+      y: 0,
+    };
 
-class DeliveryDrone {}
+    for (const value in coords) {
+      this.coords[value] = coords[value];
+    }
+
+    for (const coord in this.coords) {
+      if (coord === undefined) {
+        this.coords[coord] = 0;
+      }
+    }
+    this.chipVersion = chipVersion;
+  }
+
+  goForward(step = 1) {
+    this.coords.y += step;
+  }
+  goBack(step = 1) {
```

Comment:
Add spaces to improve the code readability
```suggestion
  }
  
  goBack(step = 1) {
```